README.md

```diff
@@ -326,6 +372,111 @@ when you want to proxy this traffic. Otherwise, it is optional. See
+
+To enable Cloud Monitoring and Cloud Trace, use the `--telemetry-project` flag
+with the project where you want to view metrics and traces. To configure the
+metrics prefix used by Cloud Monitoring, use the `--telemetry-prefix` flag. When
+enabling telementry, both Cloud Monitoring and Cloud Trace are enabled. To
+disable Cloud Monitoring, use `--disable-metrics`. To disable Cloud Trace, use
+`--disable-traces`.
+
+To enable Prometheus, use the `--prometheus` flag. This will start an HTTP
+server on localhost with a `/metrics` endpoint. The Prometheus namespace may
+optionally be set with `--prometheus-namespace`.
+
+## Frequently Asked Questions
+
+### Why would I use the proxy?
+
+The proxy is a convenient way to control access to your database using IAM
+permissions while ensuring a secure connection to your Cloud SQL instance. When
+using the proxy, you do not have to manage database client certificates,
+configured Authorized Networks, or ensure clients connect securely. The proxy
+handles all of this for you.
+
+### How should I use the proxy?
+
+The proxy is a gateway to your Cloud SQL instance. Clients connect to the proxy
+over an unencrypted connection and are authorized using the environment's IAM
+principal. The proxy then encrypts the connection to your Cloud SQL instance.
+
+Because client connections are not encrypted and authorized using the
+environment's IAM principal, we recommend running the proxy on the same VM (or
+GKE pod) as your application. This means unencrypted traffic does not leave the
+VM. It also means there is a one-to-one correspondance between the IAM principal
+used by the application and the proxy. This approach ensures a secure connection
+and leverages all the power of IAM.
+
+Here are some common examples of how to run the proxy in different environments:
+
+- [Connect to Cloud SQL for MySQL from your local computer][local-quickstart]
+- [Connect to Cloud SQL for MySQL from Google Kubernetes Engine][gke-quickstart]
+
+[local-quickstart]: https://cloud.google.com/sql/docs/mysql/connect-instance-local-computer
+[gke-quickstart]: https://cloud.google.com/sql/docs/mysql/connect-instance-kubernetes
+
+### Why can't the proxy connect to my private IP instance?
+
+The proxy does not configure the network. You MUST ensure the proxy can reach
+your Cloud SQL instance, either by deploying it in a VPC that has access to your
+Private IP instance, or by configuring Public IP.
+
+If no network path exists, the proxy cannot create a network path and so cannot
+connect.
+
+### Is there a library version of the proxy that I can use?
+
+Yes. Cloud SQL supports three language connectors:
+
+- [Cloud SQL Go Connector][go connector]
+- [Cloud SQL Java Connector](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory)
+- [Cloud SQL Python Connector](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector)
+
+If you're using Go, Java, or Python, the language connectors offer the best
```

Comment:
"The connectors for Go, Java, and Python offer the best experience when you are writing an application in those languages. Use the proxy when your application uses another language."